lib/pharos/configuration/host.rb

```diff
@@ -18,23 +19,85 @@ class Host < Pharos::Configuration::Struct
       attribute :taints, Pharos::Types::Strict::Array.of(Pharos::Configuration::Taint)
       attribute :user, Pharos::Types::Strict::String
       attribute :ssh_key_path, Pharos::Types::Strict::String
+      attribute :ssh_port, Pharos::Types::Strict::Integer.default(22)
       attribute :ssh_proxy_command, Pharos::Types::Strict::String
       attribute :container_runtime, Pharos::Types::Strict::String.default('docker')
       attribute :environment, Pharos::Types::Strict::Hash
-      attribute :bastion, Pharos::Configuration::Bastion
+      attribute :bastion, Pharos::Configuration::Bastion.optional.default(nil)
 
       attr_accessor :os_release, :cpu_arch, :hostname, :api_endpoint, :private_interface_address, :resolvconf, :routes, :config
 
+      # @return [String]
       def to_s
         short_hostname || address
       end
 
+      # @return [String,nil]
       def short_hostname
         return nil unless hostname
 
         hostname.split('.').first
       end
 
+      # @return [Hash]
+      def ssh_options
+        {}.tap do |opts|
+          opts[:keys] = [ssh_key_path] if ssh_key_path
+          opts[:send_env] = [] # override default to not send LC_* envs
+          opts[:proxy] = Net::SSH::Proxy::Command.new(ssh_proxy_command) if ssh_proxy_command && !bastion
+          opts[:port] = ssh_port
+        end
+      end
+
+      # @return [Pharos::Transport::Gateway]
+      def gateway
```

Comment:
Not completely happy about the idea that `Host` is becoming a "god" object that knows everything. For example `gateway` requires `host` and `Host` requires `Gateway` because of `kube_client` 🤔 /cc @jnummelin

Reply:
Probably all ssh specific stuff should be under Transport (somehow).

Reply:
Yes, I agree. Now `Host`, `Bastion`, `Gateway` all entangle pretty badly and really hard to follow which is used where.

Reply:
I moved stuff around.

Now there's `Pharos::Kube::Client` that understands `Configuration::Host`, so that host can just do `@kube_client ||= Pharos::Kube::Client.new(self)`.

The host ssh options are generated in `Pharos::Transport::SSH.options_for(host)`.

The `Host` now has:
- `def gateway`
- `def transport`
- `def kube_client`
- `def disconnect`

Reply:
Alternative to having these memoized in Host I guess would be to go back to 
`Pharos::SshManager.clients[host.address]` / `Pharos::GatewayManager.gateways[host.address]` type of connection collections or maybe have some sort of `Pharos::Transport` delegator/decorator and just have `host.transport` and all those `kube_client / exec / file / gateway`  stuff would be contained there.

Reply:
IMO the latter sounds pretty good

Reply:
I still think that `gateway` should be masked behind `transport`. Transport already gets host (like Gateway) as a parameter.

Reply:
@kke @jnummelin are you fine with `gateway` being here?

Reply:
I'm now WIPping with https://github.com/kontena/net-ssh-proxy-gateway